hoodie-client/src/main/java/com/uber/hoodie/func/payload/HoodieRecordBufferedIteratorPayload.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.func.payload;
+
+import com.uber.hoodie.common.model.HoodieRecord;
+import com.uber.hoodie.common.model.HoodieRecordPayload;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+
+public class HoodieRecordBufferedIteratorPayload<T extends HoodieRecordPayload>
+    extends AbstractBufferedIteratorPayload<HoodieRecord<T>, Optional<IndexedRecord>> {
+
+  // It caches the exception seen while fetching insert value.
+  public Optional<Exception> exception = Optional.empty();
+
+  public HoodieRecordBufferedIteratorPayload(HoodieRecord record, Schema schema) {
```

Comment:
I think we should consider using a static method lambda if possible to capture the transform.. This whole class structure with constructor chaining/inheritance all seem overkill for the amount of work thats done in converting I to O. This is one of the reasons function passing was introduced in java - to avoid such boilerplate code. Can we take adv of that?

Reply:
Refactored, please take a pass.